# HG changeset patch
# User Rik <rik@octave.org>
# Date 1346632903 25200
#      Sun Sep 02 17:41:43 2012 -0700
# Node ID f4b5304ee4514187f87d02eb9b97fab42ef88d58
# Parent  e359158c917f6b2cb694b52b5999aedbb6167907
build: Remove unnecessary -I includes from AM_CPPFLAGS.

* libinterp/Makefile.am: Remove unnecessary -I includes from AM_CPPFLAGS.

* src/Makefile.am: Remove unnecessary -I includes from AM_CPPFLAGS.

* liboctave/Makefile.am: Use $top_builddir to make list more readable.

* liboctave/cruft/Makefile.am: Replace tabs with spaces in file list.

* liboctave/cruft/slatec-fn/module.mk: Add blank line at end of file.

diff --git a/libinterp/Makefile.am b/libinterp/Makefile.am
--- a/libinterp/Makefile.am
+++ b/libinterp/Makefile.am
@@ -20,22 +20,21 @@
 
 include $(top_srcdir)/build-aux/common.mk
 
 AUTOMAKE_OPTIONS = subdir-objects
 
 ## Search local directories before those specified by the user.
 AM_CPPFLAGS = \
   -I$(top_srcdir)/liboctave/cruft/misc \
-  -I../liboctave/array -I$(top_srcdir)/liboctave/array \
-  -I../liboctave/numeric -I$(top_srcdir)/liboctave/numeric \
-  -I../liboctave/operators -I$(top_srcdir)/liboctave/operators \
-  -I../liboctave/system -I$(top_srcdir)/liboctave/system \
-  -I../liboctave/util -I$(top_srcdir)/liboctave/util \
-  -I../liboctave -I$(top_srcdir)/liboctave \
+  -I$(top_srcdir)/liboctave/array \
+  -I$(top_builddir)/liboctave/numeric -I$(top_srcdir)/liboctave/numeric \
+  -I$(top_builddir)/liboctave/operators -I$(top_srcdir)/liboctave/operators \
+  -I$(top_srcdir)/liboctave/system \
+  -I$(top_srcdir)/liboctave/util \
   -I$(srcdir)/octave-value \
   -I$(srcdir)/operators \
   -Iparse-tree -I$(srcdir)/parse-tree \
   -Iinterp-core -I$(srcdir)/interp-core \
   -Iinterpfcn -I$(srcdir)/interpfcn \
   -Icorefcn \
   -I$(top_builddir)/libgnu -I$(top_srcdir)/libgnu
 
@@ -165,17 +164,17 @@ liboctinterp_la_CPPFLAGS = @OCTINTERP_DL
 include link-deps.mk
 
 liboctinterp_la_LIBADD = \
   octave-value/liboctave-value.la \
   parse-tree/libparse-tree.la \
   interp-core/libinterp-core.la \
   interpfcn/libinterpfcn.la \
   corefcn/libcorefcn.la \
-  ../liboctave/liboctave.la \
+  $(top_builddir)/liboctave/liboctave.la \
   $(LIBOCTINTERP_LINK_DEPS)
 
 # Increment these as needed and according to the rules in the libtool manual:
 liboctinterp_current = 1
 liboctinterp_revision = 1
 liboctinterp_age = 0
 
 liboctinterp_version_info = $(liboctinterp_current):$(liboctinterp_revision):$(liboctinterp_age)
@@ -222,19 +221,20 @@ DLL_CXXDEFS = @OCTINTERP_DLL_DEFS@
 ## though we don't want it.  It would be super awesome if automake
 ## would allow users to choose the header file extension.
 .yy.cc:
 	$(am__skipyacc) $(SHELL) $(YLWRAP) $< y.tab.c $@ y.tab.h $*.h y.output $*.output -- $(YACCCOMPILE)
 
 ## Special rules:
 ## Mostly for sources which must be built before rest of compilation.
 
-## oct-conf.h must depend on Makefile.  Calling configure
-## may change default/config values.  However, calling configure will also
-## regenerate the Makefiles from Makefile.am and trigger the rules below.
+## oct-conf.h must depend on Makefile.
+## Calling configure may change default/config values.
+## However, calling configure will also regenerate the Makefiles from
+## Makefile.am and trigger the rules below.
 oct-conf.h: oct-conf.in.h Makefile
 	@$(do_subst_config_vals)
 
 version.h: version.in.h Makefile
 	$(SED) < $< \
 	  -e "s|%NO_EDIT_WARNING%|DO NOT EDIT!  Generated automatically from $(<F) by Make.|" \
 	  -e "s|%OCTAVE_API_VERSION_NUMBER%|${OCTAVE_API_VERSION_NUMBER}|" \
 	  -e "s|%OCTAVE_API_VERSION%|\"${OCTAVE_API_VERSION}\"|" \
diff --git a/liboctave/Makefile.am b/liboctave/Makefile.am
--- a/liboctave/Makefile.am
+++ b/liboctave/Makefile.am
@@ -95,17 +95,17 @@ include link-deps.mk
 liboctave_la_LIBADD = \
   array/libarray.la \
   cruft/libcruft.la \
   cruft/ranlib/libranlib.la \
   numeric/libnumeric.la \
   operators/liboperators.la \
   system/libsystem.la \
   util/libutil.la \
-  ../libgnu/libgnu.la \
+  $(top_builddir)/libgnu/libgnu.la \
   $(LIBOCTAVE_LINK_DEPS)
 
 # Increment these as needed and according to the rules in the libtool manual:
 liboctave_current = 1
 liboctave_revision = 1
 liboctave_age = 0
 
 liboctave_version_info = $(liboctave_current):$(liboctave_revision):$(liboctave_age)
diff --git a/liboctave/cruft/Makefile.am b/liboctave/cruft/Makefile.am
--- a/liboctave/cruft/Makefile.am
+++ b/liboctave/cruft/Makefile.am
@@ -62,11 +62,11 @@ include slatec-fn/module.mk
 ## Special rules for files which must be built before compilation
 cruft.def: $(libcruft_la_SOURCES) mkf77def
 	chmod a+rx mkf77def
 	./mkf77def $(srcdir) $(libcruft_la_SOURCES) > $@-t
 	mv $@-t $@
 
 DISTCLEANFILES = \
   cruft.def \
-	ranlib/ranlib.def
-	$(nodist_libcruft_la_SOURCES)
+  ranlib/ranlib.def \
+  $(nodist_libcruft_la_SOURCES)
 
diff --git a/liboctave/cruft/slatec-fn/module.mk b/liboctave/cruft/slatec-fn/module.mk
--- a/liboctave/cruft/slatec-fn/module.mk
+++ b/liboctave/cruft/slatec-fn/module.mk
@@ -73,8 +73,9 @@ nodist_libcruft_la_SOURCES += \
 
 slatec-fn/erfc.f: slatec-fn/erfc.in.f Makefile
 	$(SED) -e "${F77_ISNAN_MACRO}" < $< > $@-t
 	mv $@-t $@
 
 slatec-fn/derfc.f: slatec-fn/derfc.in.f Makefile
 	$(SED) -e "${F77_ISNAN_MACRO}" < $< > $@-t
 	mv $@-t $@
+
diff --git a/src/Makefile.am b/src/Makefile.am
--- a/src/Makefile.am
+++ b/src/Makefile.am
@@ -17,22 +17,21 @@
 # You should have received a copy of the GNU General Public License
 # along with Octave; see the file COPYING.  If not, see
 # <http://www.gnu.org/licenses/>.
 
 include $(top_srcdir)/build-aux/common.mk
 
 ## Search local directories before those specified by the user.
 AM_CPPFLAGS = \
-  -I$(srcdir)/../liboctave \
-  -I$(srcdir)/../liboctave/array \
-  -I$(srcdir)/../liboctave/cruft/misc \
-  -I$(srcdir)/../liboctave/numeric \
-  -I$(srcdir)/../liboctave/util \
-  -I$(srcdir)/../libinterp \
+  -I$(top_srcdir)/liboctave/array \
+  -I$(top_srcdir)/liboctave/cruft/misc \
+  -I$(top_srcdir)/liboctave/numeric \
+  -I$(top_srcdir)/liboctave/util \
+  -I$(top_srcdir)/libinterp \
   -I$(top_builddir)/libinterp/interpfcn \
   -I$(top_builddir)/libgnu -I$(top_srcdir)/libgnu
 
 OCTAVE_BINARIES = \
   octave \
   octave-cli
 
 if AMCOND_BUILD_COMPILED_AUX_PROGRAMS
@@ -66,34 +65,34 @@ endif
 
 EXTRA_DIST = \
   mkoctfile.in.cc \
   mkoctfile.in.sh \
   octave-config.in.cc \
   octave-config.in.sh
 
 OCTAVE_CORE_LIBS = \
-  ../libinterp/liboctinterp.la \
-  ../liboctave/liboctave.la
+  $(top_builddir)/libinterp/liboctinterp.la \
+  $(top_builddir)/liboctave/liboctave.la
 
 include ../libgui/src/link-deps.mk
 
 octave_cli_SOURCES = main-cli.cc
 
 octave_cli_LDADD = \
   $(OCTAVE_CORE_LIBS)
   $(OCTAVE_LINK_DEPS)
 
 octave_cli_LDFLAGS = \
   $(NO_UNDEFINED_LDFLAG) \
   $(OCTAVE_LINK_OPTS)
 
 if AMCOND_BUILD_GUI
   octave_SOURCES = main.cc
-  OCTAVE_GUI_LIBS = ../libgui/src/liboctgui.la
+  OCTAVE_GUI_LIBS = $(top_builddir)/libgui/src/liboctgui.la
   OCTAVE_GUI_CPPFLAGS = -I$(top_srcdir)/libgui/src
 else
   octave_SOURCES = main-cli.cc
   OCTAVE_GUI_LIBS =
   OCTAVE_GUI_CPPFLAGS =
 endif
 
 octave_CPPFLAGS = \
